src/common/AndroidUtils.ts

```diff
@@ -533,33 +536,19 @@ export class AndroidUtils {
     public static async waitUntilDeviceIsReady(
         portNumber: number
     ): Promise<void> {
-        const command = `${AndroidUtils.getAdbShellCommand()} -s emulator-${portNumber} wait-for-device shell getprop sys.boot_completed`;
+        const quote = process.platform === WINDOWS_OS ? '"' : "'";
+        const bootCmd = `shell ${quote}while [[ -z $(getprop sys.boot_completed) ]]; do sleep 1; done;${quote}`;
```

Comment:
Read through this [SO post](https://stackoverflow.com/questions/3634041/detect-when-android-emulator-is-fully-booted) and all of the various ways of detecting when AVD is booted. `getprop sys.boot_completed` still seems to be the best approach.

But rather than doing the loop/sleep/poll on host machine, I've modified it to do that on the AVD itself.